pkg/pillar/cas/cas.go

```diff
@@ -0,0 +1,66 @@
+package cas
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+//CAS: provides methods to interact with CAS clients
+type CAS interface {
+	//Blob APIs
+	//CheckBlobExists: returns true if the blob exists.
+	CheckBlobExists(blobSha256 string) bool
+	//GetBlobSize: returns size of the blob
+	GetBlobSize(blobSha256 string) (int64, error)
+	//ListBlobs: returns list of blob sha256
+	ListBlobs() ([]string, error)
+	//IngestBlob: ingests a blob from the given reader and assigns the given sha256 to it.
+	IngestBlob(blobSha256 string, reader io.Reader) error
+	//ReadBlob: returns a reader of blob
+	ReadBlob(sha256 string) (io.Reader, error)
+	//RemoveBlob: removes the blob after the given timestamp.
+	RemoveBlob(blobSha256 string, timeStamp time.Time) (*time.Timer, error)
```

Comment:
Same as the above with argument naming, and giving it the correct description of `<algo>:<hash>`

Reply:
Actually, I am a bit concerned about this function. As of now, we are mostly sticking with the "let containerd manage it" garbage collection strategy. That means that our ability to do "remove it in future" really doesn't exist, as other things hold onto it. Our only choices are:

* let containerd GC it any way it wants
* remove it forcibly (i.e. _right now_)

I think we might want to leave this off for now, add it in the future. Thoughts?

Reply:
Done.

Reply:
Done.

Reply:
>I think we might want to leave this off, for now, add it in the future. Thoughts? 

Yeah. Since we already specify a lease duration, I guess "delete in future" part is not needed here as anyway if the blob is not referred by anything, it'll be GCed after the lease expires.